### DISCUSSION TITLE
Create log from flows results in error: "An Apex error occurred: System.SObjectException: Invalid field OriginLocation__c"

### DISCUSSION THREAD

**Discussion by dwever** [jongpie/NebulaLogger#558] [Q&A]
Creating a log from flow results in following error: "Error Occurred: An Apex error occurred: System.SObjectException: Invalid field OriginLocation__c"

What is not correct with the input parameters? See below

Creating a log from flow with following Input:
FLOWLOGENTRY (APEX): Log Entry
Inputs:
faultMessage = {!forT_Log_FlowFaultErrorMessage} (You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. --- You don't have permission to do this. ) --> this is the fault message
flowName = {!forT_Log_FlowAPIName} (ADMIN_Autolaunched_Delete_Records) --> this is the api name of the flow
loggingLevelName = {!$Setup.LoggerSettings__c.LoggingLevel__c} (FINEST)
message = {!forT_Log_LogEntryMessage} (Flow | ADMIN - Autolaunched - Delete FieldPro Records)
recordId = {!forT_Log_RecordId} (12345567) --> This is a RecordId of a user
saveLog = true
saveMethodName = {!$Setup.LoggerSettings__c.DefaultSaveMethod__c} (EVENT_BUS)
scenario = {!$Setup.LoggerSettings__c.DefaultScenario__c} (Log)

**Comment by jongpie** [comment]
Hi @dwever - I've never seen this error before, I'm not sure what the cause could be. Do you know which version of Nebula Logger you're using in your org? And can you check to confirm if the `OriginLocation__c` fields exist in your org? There are 2 fields with that name:
- `LogEntryEvent__e.OriginLocation__c` - it should show up under Setup --> Platform Events --> Log Entry Event
- `LogEntry__c.OriginLocation__c` - it should show up under Setup --> Object Manager --> Log Entry --> Fields & Relationships

**Reply by dwever** [reply]
@jongpie : I'm using version 4.11.5.

Both fields with the OriginLocation__c are available in the org. Can you elaborate on how these fields are populated? Might help me with root cause analysis. Is this field dependent on the input parameters used when creating a log from flow?

**Reply by jongpie** [reply]
Very strange! `OriginLocation__c` is automatically set (via Apex code) when the Flow invocable actions are used. Given that the fields are present in your org, I think there's some other issue at hand, and the error saying it's an invalid field isn't the actual issue here.

Just a couple of follow-up questions for you:

- When the Flow error occurs, is that happening for all users (including system admin users)? Or does it only happen for certain users? I'm wondering if permissions or licenses could be a part of the problem
- Are you able to log using Apex? You can try running this anonymous Apex code in developer console - if everything works, you'll have a new `Log__c` record generated:
   ```apex
   Logger.info('hello, world');
   Logger.saveLog();
   ```
